cpg_infra/driver.py

```diff
@@ -661,38 +657,33 @@ def setup_externally_specified_members(self):
 
         for group in groups:
             group_name = group.name.removeprefix(self.dataset_config.dataset + '-')
-            for member in self.dataset_config.members.get(group_name, []):
-                h = self.compute_hash(self.dataset_config.dataset, member)
-
-                # always add all GCP members, require them to exist in the map for Azure
-                skip_members_without_id = True
-                if isinstance(self.infra, GcpInfrastructure):
-                    member_map = {}
-                    skip_members_without_id = False
-                elif isinstance(self.infra, AzureInfra):
-                    member_map = self.config.member_to_azure_account
-                else:
-                    raise ValueError(f'No member map for {self.infra.name()}')
-
-                if cloud_id := member_map.get(
-                    member, None if skip_members_without_id else member
-                ):
+            for member_id in self.dataset_config.members.get(group_name, []):
+                member = self.config.users.get(member_id)
+                if not member:
+                    raise ValueError(f'Member {member_id} not found in config')
+
+                h = self.compute_hash(
+                    dataset=self.dataset_config.dataset,
+                    member=member.id,
+                    cloud=self.infra.name(),
+                )
+                if cloud_user := member.clouds[self.infra.name()]:
```

Comment:
Could this raise a `KeyError` if the infra name is not in a member's clouds dict, or is `member.clouds` is initialised to contain all expected infra names?

Reply:
We only want to add someone to a cloud where we know the "id" we can use to reference them. In Google Cloud this is easy, it's just their email (encoded as gcp.id in the user.yaml). For azure, this is a specific object ID they get when they're guest-ed into our tenant. So we only want to add them if we know that ID. 

It could lead to some awkward debugging, "why isn't this person in the azure group", but I don't want to add logging because it's the more common case that they're not added to an equivalent azure resource. 

Same vibes for Hail below, only add them if we know they're hail ID for the specific cloud.